### DISCUSSION TITLE
MQTT cmd number brightness

### DISCUSSION THREAD

**Discussion by DandeMC** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#269] [Ideas]
Hello,

is there any cmd number to set the brightness over mqtt? If not, could that still be included?

**Comment by Skeletorjus** [comment]
That would actually be really nice. On a related note I've also been thinking it would be cool to have a way to send text to the display via MQTT.

edit: and a way to send beeps! The combination of sound and text would actually be quite useful for automations/notifications.

**Reply by DandeMC** [reply]
It would be funny. But how many letters can it display? 3 oder 4?

The next thing I noticed is that the firmware is displayed as the date via mqtt and not the firmware number. is that correct or a bug?

**Reply by Skeletorjus** [reply]
The display can show 3 symbols, but it can scroll on longer string as seen during bootup where it shows the IP and version.
If you're on the development branch the date is used for version numbering.

**Reply by DandeMC** [reply]
I've installed 3.2.1 yesterday and on mqtt it shows 2022-04-11 is firmware. Is this version still on development branch?

**Reply by Skeletorjus** [reply]
I see now that both development and master are using date as version, so it is date in any case. You can also see this if you look at the bottom of the web interface.

**Reply by DandeMC** [reply]
Yes, there is also the date. Is it possible to switch this to the version number?

**Comment by visualapproach** [comment]
MQTT brightness was discussed in another thread and will probably happen.
Text and sound would be possible. I'll put in on the todo list.

Regarding version numbering - v3.2.1 is a github tag. The compiler is not aware of that. Current workflow is: new code uploads to development branch. I manually update the FW_VERSION to date of upload. When I think it's time, I merge dev to master branch and attach a new v3.2.2 tag in github. If I were to name it this way in the code I would need to make changes directly to master branch which will get out of sync with the dev branch. Merging two branches where both have changes is possible but not fun. Maybe someone with black belt in githubbing can come up with a better workflow.

**Reply by DandeMC** [reply]
I found this in websocket.js

	setTarget: 0,
	toggleUnit: 1,
	toggleBubbles: 2,
	toggleHeater: 3,
	togglePump: 4,
	//resetq: 5,
	restartEsp: 6,
	//gettarget: 7,
	resetTotals: 8,
	resetTimerChlorine: 9,
	resetTimerFilter: 10,
	toggleHydroJets: 11,
	setBrightness: 12
	
There is already a cmd to set the brightness. I testet it and it worked fine =)

**Reply by visualapproach** [reply]
Haha I forgot and hadn't looked into it yet. Less work then. 😁 thx
